proto/ibc/core/channel/v1/channel.proto

```diff
@@ -65,6 +65,12 @@ enum State {
   // A channel has been closed and can no longer be used to send or receive
   // packets.
   STATE_CLOSED = 4 [(gogoproto.enumvalue_customname) = "CLOSED"];
+  // A channel has just started the channel upgrade handshake.
+  // The chain that is proposing the upgrade should set the channel state from OPEN to UPGRADE_INIT.
+  STATE_UPGRADE_INIT = 5 [(gogoproto.enumvalue_customname) = "UPGRADE_INIT"];
+  // A channel has acknowledged the upgrade handshake step on the counterparty chain.
+  // The counterparty chain that accepts the upgrade should set the channel state from OPEN to UPGRADE_TRY.
+  STATE_UPGRADE_TRY = 6 [(gogoproto.enumvalue_customname) = "UPGRADE_TRY"];
```

Comment:
Protobuf defined enums are perfect! Maybe we should remove the underscore between "UPGRADE" and "INIT"/"TRY".

See here for the revised updates to what we should use for `gogoproto.enumvalue_customname`

https://github.com/cosmos/ibc/pull/803

This is essentially to align with the same naming convention used for `STATE_TRYOPEN` / `TRYOPEN` enum.